cpp/include/cudf/utilities/error.hpp

```diff
@@ -58,13 +58,13 @@ struct cuda_error : public std::runtime_error {
  * @param[in] reason String literal description of the reason that cond is
  * expected to be true
  * @throw cudf::logic_error if the condition evaluates to false.
- **/
-#define CUDF_EXPECTS(cond, reason)                                  \
-  (!!(cond)) ? static_cast<void>(0)                                 \
-             : throw cudf::logic_error("cuDF failure at: " __FILE__ \
-                                       ":" CUDF_STRINGIFY(__LINE__) ": " reason)
-
-/**
+ *---------------------------------------------------------------------------**/
+#define CUDF_EXPECTS(cond, reason) \
+  (!!(cond))                       \
+    ? static_cast<void>(0)         \
+    : throw cudf::logic_error(     \
+        std::string("cuDF failure at: " __FILE__ ":" CUDF_STRINGIFY(__LINE__) ": ") + reason)
```

Comment:
@aschaffer I think you wanted this macro to work with string objects, instead of just string literals. (sorry if this is a mix-up)
Does this change work for you?

Reply:
I'm not sure what's going on. I haven't been involved in this PR.

Reply:
@vuule , can you please explain?

Reply:
If `reason` can now be a `std::string` then, yes, this would address some old concerns.